images/cfw/opt/x1plusd.py

```diff
@@ -1,116 +1,282 @@
+            )
+            self.build_info = {
+                "cfwVersion": "0.1",
+                "date": "2024-04-17",
+                "buildTimestamp": 1713397465.0,
+            }
+
+    async def task(self):
+        # Catch our OTA calls
+        OTAMatch = MatchRule(
+            interface="x1plus.ota", path="/x1plus/ota", type="method_call"
+        )
+        await Proxy(message_bus, self.router).AddMatch(OTAMatch)
+        with self.router.filter(OTAMatch, bufsize=0) as queue:
+            while True:
+                msg = await queue.get()
+                method = msg.header.fields[HeaderFields.member]
+
+                if msg.header.fields[HeaderFields.signature] != "s":
+                    await self.router.send(
+                        new_error(msg, "x1plus.x1plusd.Error.BadSignature")
+                    )
+                    continue
+
+                arg = msg.body[0]
+
+                rv = None
+                try:
+                    if method == "ota":
+                        rv = await self._update_check(json.loads(arg))
+
+                except Exception as e:
+                    logger.error(f"{method}({arg}) -> exception {e}")
+                    await self.router.send(
+                        new_error(msg, "x1plus.x1plusd.Error.InternalError")
+                    )
+                    continue
+
+                if not rv:
+                    logger.warning(f"{method} -> NoMethod")
+                    await self.router.send(
+                        new_error(msg, "x1plus.x1plusd.Error.NoMethod")
+                    )
+                    continue
+
+                logger.debug(f"{method}({arg}) -> {rv}")
+                await self.router.send(new_method_return(msg, "s", (rv,)))
+
+    async def _update_check(self, payload):
+        ota_response = {
+            "ota_available": self.ota_available,
+            "err_on_last_check": self.last_check_error,
+            "last_checked": self.last_check_timestamp,
+            "ota_info": self.last_check_response,
+            "is_downloaded": self.ota_downloaded,
+        }
+
+        # First, if we were asked just for the last status, just return our ota object
+        if not payload.get("check", False):
+            return json.dumps({"ota": ota_response})
+
+        # Load in our settings.json, we need to make sure we are still enabled! :)
+        try:
+            with open(
+                f"/mnt/sdcard/x1plus/printers/{_get_sn()}/settings.json", "r"
+            ) as fh:
+                x1p_settings = json.load(fh)
+        except FileNotFoundError:
+            logger.warning("OTA can't find settings.json, assuming OTAs are disabled!")
+            x1p_settings = {"ota.enable": False}
+
+        # Do we have OTAs enabled? If not, just return current status
+        if not x1p_settings.get("ota.enable", False):
+            logger.info("OTA check is disabled, skipping check!")
+            return json.dumps({"ota": ota_response})
```

Comment:
We should 'know' this -- just reach into the settings daemon's state to ask how it's feeling.

Reply:
I mean, unless we wanna dbus across within the same app, feels dumb, esp since in our initial draft doc we said we should trust the settings file as the source of truth.

However, if you prefer this is dbus so we ensure ALL things go through the settings dbus interface, im not against that. :)

Reply:
Well, what I mean by this is that we are in-process.  So you could, say, just ask the `SettingsService.settings` or something, no need for any DBus-to-DBus.  Right now that's hidden in `settings` but presumably we could have a `x1plusd` object that holds all of this.